Automattic/Gravatar-SDK-Android
# QuickEditor: Avatar upload error UI

Closes #267 

### Description

This PR adds an option to retry failed uploads by keeping those failed tries in the state and displaying them with an error indicator. The dialog with `remove upload` and `retry` options is shown when tapped.

I removed the `LocalAvatar` composable and put the "error" overlay in the `SelectableAvatar`. 
I added `AvatarLoadingState` sealed class to make the state modeling easier instead of a bunch of Bool values. 

### Testing Steps

2. Run the DemoApp 
3. Go to Avatar Update tab
4. Tap `Update Avatar` button
5. Follow the steps to log in (if necessary)
11. Enable request mocking for post requests or change the aspect ratio in the UCropCropperLauncher
12. Pick or take a photo and upload it
13. Confirm you see an error overlay
14. Tap it
15. Retry
16. Confirm you see an error overlay
17. Tap it
18. Remove upload
19. Confirm it's gone
20. Disable request mocking
